### PR TITLE
fix: Remove undefined from events in update

### PR DESCRIPTION
**Pull request by olivermrbl**
**What**

Let's imagine we perform a selector update on a resource that doesn't exist:
```
await updateCountries({ region_id: "reg_123" }, { : metadata: { ... } })
```
(`reg_123` doesn't exist). 

The underlying Medusa Service `update` call will return an empty result set. Because we assign the first entry in the result set to `response,` the response will be `[ undefined ]`.

Then, `response` is passed to the event factory builder that builds events using `response.map(r => r.id)` which is the piece that throws and leads to the following error: 
```
message: "Cannot read properties of undefined (reading 'id')",
stack: "TypeError: Cannot read properties of undefined (reading 'id')\n at /[redacted]/node_modules/@medusajs/utils/src/modules-sdk/event-builder-factory.ts:69:30\n"
```

@adrien2p @carlos-r-l-rodrigues let me know if this change is better suited elsewhere.

### CODE REVIEW

File: packages/modules/region/integration-tests/__tests__/region-module.spec.ts [hunk 1]
```diff
@@ -403,6 +403,16 @@ moduleIntegrationTestRunner<IRegionModuleService>({
         expect(resp.countries).toHaveLength(2)
       })
 
+      it.only("should delete a region without countries", async () => {
```
Comment (adrien2p) [inline packages/modules/region/integration-tests/__tests__/region-module.spec.ts]: Todo: rm .only

**Review by adrien2p — APPROVED** [review]
Nice catch 🫡

File: packages/modules/region/integration-tests/__tests__/region-module.spec.ts [hunk 2]
```diff
@@ -403,6 +403,16 @@ moduleIntegrationTestRunner<IRegionModuleService>({
...
       })
 
+      it("should delete a region without countries", async () => {
+        const createdRegion = await service.createRegions({
+          name: "North America",
+          currency_code: "USD",
+          countries: [],
+        })
+
+        await service.softDeleteRegions([createdRegion.id])
```
Comment (adrien2p) [inline packages/modules/region/integration-tests/__tests__/region-module.spec.ts:413]: Q: should we add an expectation ? You can retrieve with deleted and check that it is properly set

Comment (olivermrbl) [reply packages/modules/region/integration-tests/__tests__/region-module.spec.ts:413]: Yeah, just thought of it, let me do that 😄

**Review by carlos-r-l-rodrigues — APPROVED** [review]

**Pull request merged**